File: Python/2941번 크로아티아 알파벳.py
def main():
    maps = {
        "c=": 1,
        "c-": 1,
        "dz=": 1,
        "d-": 1,
        "lj": 1,
        "nj": 1,
        "s=": 1,
        "z=": 1
    }

    s = input()
    count = 0
    i = 0

    while i < len(s):
        if s[i:i+3] in maps:
            count += 1
            i += 3
        elif s[i:i+2] in maps:
            count += 1
            i += 2
        else:
            count += 1
            i += 1

    print(count)


if __name__ == "__main__": 
    main()

File: C++/2941번 크로아티아 알파벳.cpp
#include<iostream>
#include<vector>
#include<map>
using namespace std;
map<string,int>maps ={
 {"c=",1},
 {"c-",1},
 {"dz=",1},
 {"d-",1},
 {"lj",1},
 {"nj",1},
 {"s=",1},
 {"z=",1}
};

int main(){
string str;
cin >> str;
int count =0;
for(int i=0;i<str.size();i++){
    if(maps.find(str.substr(i,3))!=maps.end()){
        count ++;
        i+=2;
    }
    else if(maps.find(str.substr(i,2))!=maps.end()){
        count++;
        i++;
    }
    else{
    count ++;}
}
cout << count;


return 0;

}